src/Collection/Collection.router.ts

```diff
@@ -333,96 +335,92 @@ export class CollectionRouter extends Router {
 
   upsertCollection = async (req: AuthRequest): Promise<FullCollection> => {
     const id = server.extractFromReq(req, 'id')
-    const collectionJSON: FullCollection = server.extractFromReq(
-      req,
-      'collection'
-    )
-    const data: string = server.extractFromReq(req, 'data')
+    let collectionJSON: FullCollection
+    try {
+      collectionJSON = server.extractFromReq(req, 'collection')
+    } catch (error) {
+      throw new HTTPError(
+        (error as Error).message,
+        null,
+        STATUS_CODES.badRequest
+      )
+    }
```

Comment:
Are we doing this to avoid a 500 error? I ask because the very same thing is happening with all other `server.extractFromReq` calls. Say `server.extractFromReq(req, 'data')` below for example

also you can avoid the null by doing `new HTTPError(error.message).setStatusCode(STATUS_CODES.badRequest)` as it returns `this`